### ISSUE TITLE
Option to have MessageTemplateTextFormatter escape CRLF.

### ISSUE THREAD

**Issue by aradford** [serilog/serilog#1147]
New Feature request.

I am using Serilog to write to a (text) log file that will be parsed with another program. I would like to output one line for each event. That means I do not want the message or exception to contain CRLFs.

As far as I can tell, the is not way to have MessageTemplateTextFormatter escape CRLFs into \r\n. 

A possible implementation for this would involve a new modifier, possibly :n, that would escape newlines.

Or maybe there is an existing feature that will do this that I don't know about.

**Comment by nblumhardt** [comment]
Hi Andrew,

It sounds like implementing `ITextFormatter` will be the way to go, for you - the text-based sinks all accept one of these, and it's a reasonably straightforward interface to implement for turning a `LogEvent` into a string.

How does this look for you?

Cheers,
Nick

**Comment by aradford** [comment]
Thanks for the response.

Implementing ITextFormatter could work, but if I want to be able to have a configurable outputTemplate property, I would be duplicating a lot of what's in MessageTemplateTextFormatter. I will try implementing something and see if I discover any way to do it without too much duplication.

**Comment by aradford** [comment]
Actually I was able to get the results that I wanted. I was able to solve the message problem by changing "{Message}" to "{Message:j}" in the outputTemplate.

I changed the exception by creating an enricher that adds a property called EscapedException, which is just logEvent.Exception.ToString().Replace("\r\n", "\\r\\n"). Then I replaced "{Exception}" with "{EscapedException}" in the outputTemplate.

[Link to Stack Overflow question that I had posted.](https://stackoverflow.com/questions/49700265/remove-new-lines-from-message-and-exception)
